### DISCUSSION TITLE
better mesh control ACL style

### DISCUSSION THREAD

**Discussion by realitytech** [meshtastic/firmware#3571] [Ideas]
Would is be possible to add in some form of access control for nodes, at the very least a good ignore list. I know it goes against the idea but we have built a large local mesh of 50ish nodes. 
All is good until someone enables MQTT on LongFast and everything breaks. 
We've been reasonably sucessful on educating peopl ebut we now have one node we cant's seen to get contact with spewing vast amounts of MQTT traffic. It would be handy if we could at least push out a stop to our routers.

**Comment by pista01** [comment]
There needs to be an intelligent router between meshes on MQTT.  Stop routing everything and only have "registered" nodes participate.  A proper mesh router should be able to handle multiple MQTT endpoints with the device registration indicating allowed mesh to mesh on a per node basis.  Node-Red would be a good place to build this.

**Reply by realitytech** [reply]
The issue is people are activeley turning MQTT on in public channels :( I can think of a dozen use cases for MQTT in a private space, hell I plan on using it on some jobs in place of wifi and using MQTT to get it back to head office, but this is a small private mesh. I dont need to seen nodes on the Eifel Tower or in Salt Lake on the South coast of the UK.

Maybe a way would be to specifically opt in at a server level, eg you need to register for relay and this gives a way of controlling what goes in and out and a mechanism for stopping abuse.

**Reply by pista01** [reply]
Hence my suggestion to register each device that needs to communicate through mqtt.  Each mesh owner would need to implement this router to prevent their mesh from being spammed.  Local registered mesh nodes would be allowed to send to the public mqtt and registered remote nodes would be accepted into the local mesh.
